zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessCreatedV2Applier.java

```diff
@@ -25,5 +25,6 @@ public ProcessCreatedV2Applier(final MutableProcessingState state) {
   public void applyState(final long processDefinitionKey, final ProcessRecord value) {
     processState.putProcess(processDefinitionKey, value);
     processState.storeProcessDefinitionKeyByProcessIdAndDeploymentKey(value);
+    processState.storeProcessDefinitionKeyByProcessIdAndVersionTag(value);
```

Comment:
💭 Just commenting for clarity:

This class was added with PR #20498 so any records that are processed with this applier must be created by Zeebe `8.6.0`, i.e. they will be `versionTag`-aware.

Reply:
Yes, I figured that it would be a similar case as above where processes are stored by id and deployment key.